tools/mkbuildoptglobals.py

```diff
@@ -295,16 +299,16 @@ def copy_create_build_file(source_fqfn, build_target_fqfn):
                 pass
     return True     # file changed
 
-
 def add_include_line(build_opt_fqfn, include_fqfn):
+    global default_encoding
     if not os.path.exists(include_fqfn):
         # If file is missing, we need an place holder
-        with open(include_fqfn, 'w', encoding="utf-8"):
+        with open(include_fqfn, 'w', encoding=default_encoding):
```

Comment:
missed one odd encoding=
```suggestion
        with open(include_fqfn, 'w'):
```

Reply:
What you are suggesting is I didn't need any of this `get_encoding()` logic - tangent I went on. In the absence of the `-X utf8`, I just needed to selectively remove all the `, encoding="utf-8"` options where I needed the system encoding to dominate.

Since merged - do I push an update or need a new PR?

Reply:
> I just needed to selectively remove all the , encoding="utf-8" options where I needed the system encoding to dominate.

nothing is written to file, so no need for encoding= to possibly confuse future reader(s)
python won't re-encode stuff, this is just the thing for read / write through text io wrapper

in absence of -Xutf-8, text io selects something default aka depends on version
https://github.com/python/cpython/blob/cbd192b6ff137aec25913cc80b2bf7bf1ef3755b/Modules/_io/textio.c#L1102-L1141 (our bundled version)
https://github.com/python/cpython/blob/ddb65c47b1066fec8ce7a674ebb88b167f417d84/Modules/_io/textio.c#L1126-L1132 (3.11 on any recent installation)

utf-8 output is ok, reading cli / writing stuff for external tools should have system encoding

Reply:
> nothing is written to file, so no need for encoding= to possibly confuse future reader(s)

Just for clarification when creating `build.opt`, we are reading `utf-8` from dot-h files and writing with system encoding to `build.opt` otherwise I can get a file not found for a `-include ...` if the file path contains diacritics.

Reply:
also nit:

> diacritics

any time we have characters from codepage different from ansii e.g. `C:\Users\Максім` in cp1251
just my guess that GCC probably does not use win32 `_wfopen` and widechar, so we have to use exact codepage bytes for filepath representation so system exec for GCC gets proper paths

path and text have different code sections dealing with them, just to denote the difference

(I am gonna go read [Python PEP 529](https://peps.python.org/pep-0529/), https://vstinner.github.io/python36-utf8-windows.html series and cpython [unicodeobject.c](https://github.com/python/cpython/blob/main/Objects/unicodeobject.c))